# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/loader/WorkerLoadContext.cpp
# Commit: 5f483f0b842a
# Full Hash: 5f483f0b842afdfad82ea041256f9d8bce37f421
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-11-02 04:41:32
# Regressor Bug: 1797327
# File Overlap Count: 3
# Description:
#   Bug 1797327 - Use RefPtrs for all WorkerLoadContexts; r=asuth
#   
#   As WorkerLoadContexts now inherit from a non-CC'd loadContextBase, we have two outcomes.
#   1) we need to break cycles with ScriptLoadRequests manually, so that ScriptLoadRequests can be collected (ScriptLoadRequests must be CC'd).
#   2) we can now have refptrs to WorkerLoadContexts in the CacheLoadHandler and NetworkLoadHandler classes, and remove any remaining raw pointers to ScriptLoadRequest/WorkerLoadContext. There are cases where the NetworkLoadHandler or CacheLoadHandler might outlive the Worker Loader, so having refpointers here should help us recover in those cases.
# ==============================================================================

diff -r 7ade701781f5 -r 5f483f0b842a dom/workers/loader/WorkerLoadContext.cpp
--- a/dom/workers/loader/WorkerLoadContext.cpp	Tue Nov 01 18:03:34 2022 +0000
+++ b/dom/workers/loader/WorkerLoadContext.cpp	Tue Nov 01 18:03:34 2022 +0000
@@ -19,7 +19,9 @@
 void WorkerLoadContext::SetCacheCreator(
     RefPtr<workerinternals::loader::CacheCreator> aCacheCreator) {
   AssertIsOnMainThread();
-  mCacheCreator = new nsMainThreadPtrHolder<workerinternals::loader::CacheCreator>("WorkerLoadContext::mCacheCreator", aCacheCreator);
+  mCacheCreator =
+      new nsMainThreadPtrHolder<workerinternals::loader::CacheCreator>(
+          "WorkerLoadContext::mCacheCreator", aCacheCreator);
 }
 
 void WorkerLoadContext::ClearCacheCreator() {